File: AdvancePython/Day-2 Constructors and Methods/Task-5.py
# Task 5:
class Car:
    age = 20

    def __init__(self, make: str, model: str, year: int) -> None:
        self.make: str = make
        self.model: str = model
        self.year: int = year

    @property
    def age(self) -> int:
        current_year: int = 2026
        return current_year - self.year


my_car = Car("BMW", "M4", 2024)
print(f"Make: {my_car.make}")
print(f"Model: {my_car.model}")
print(f"Year: {my_car.year}")
print(f"Age: {my_car.age} years")

File: node.js/Day-2/Task-5.js
// Task 5

class CarAge {
    constructor(make, model, year) {
        this.make = make;
        this.model = model;
        this.year = year;
    }

    get age() {
        const currentYear = 2026;
        const calculatedAge = currentYear - this.year;
        return `Car age is ${calculatedAge} years.`;
    }

    set age(newAge) {
        this.year = newAge
    }
}

const info = new CarAge("BMW", "M5", 2024);
console.log(info.make);
console.log(info.model);
console.log(info.year);
console.log(info.age);